client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useOpenedWithBarcode.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect } from 'react';
+import { getPackQuantityCellId } from '../../../../utils';
+import { ScannedBarcode } from '../../../../types';
+import { useOutbound } from '../../../api';
+import { useAllocationContext } from '../allocation/useAllocationContext';
+
+export const useOpenedWithBarcode = (barcode: ScannedBarcode | null) => {
+  const { mutateAsync: insertBarcode } = useOutbound.utils.barcodeInsert();
+
+  const { itemId, draftLines } = useAllocationContext(state => ({
+    draftLines: state.draftLines,
+    itemId: state.initialisedForItemId,
+  }));
+
+  useFocusNumberOfPacksInput(barcode?.batch, itemId);
+
+  const saveBarcode = async (itemId: string) => {
+    // ID means barcode has already been saved
+    if (!barcode || barcode.id) return;
+
+    // Find pack size of first allocated line to associate with barcode
+    // Usually, scanned barcode would match only one batch, so only one line would
+    // match here - but if many matched, we will just take the first
+    const packSize = draftLines.find(line => line.numberOfPacks > 0)?.packSize;
+
+    return insertBarcode({
+      input: {
+        gtin: barcode.gtin,
+        itemId,
+        packSize,
+      },
+    });
+  };
+
+  return { saveBarcode };
+};
+
+const useFocusNumberOfPacksInput = (
+  batch: string | undefined,
+  itemId: string | null
+) => {
+  useEffect(() => {
+    if (!batch || !itemId) return;
+    setTimeout(() => {
+      const input = document.getElementById(getPackQuantityCellId(batch));
```

Comment:
I guess this is copy and paste ? useRef set to zustand state would be the alternative

Reply:
Yeah this was existing...